packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -338,23 +365,32 @@ async function validateGrant(
   trx: TransactionOrKnex,
   callback?: (f: unknown) => NodeJS.Timeout,
   grantLockTimeoutMs: number = 5000
-): Promise<boolean> {
+): Promise<GrantValidationResult> {
   if (!grant.limits) {
-    return true
+    return { isValid: true }
   }
   const paymentLimits = validateAccessLimits(payment, grant.limits)
   if (!paymentLimits) {
-    return false
+    return { isValid: false }
+  }
+
+  let paymentLimitAmount: Amount
+
+  if (hasDebitAmount(paymentLimits)) {
+    paymentLimitAmount = paymentLimits.debitAmount
+  } else if (hasReceiveAmount(paymentLimits)) {
+    paymentLimitAmount = paymentLimits.receiveAmount
+  } else {
+    return { isValid: true }
   }
```

Comment:
I think we can only do this once
- https://github.com/interledger/open-payments/issues/471

is merged? Since now, we _can_ have both

Reply:
I renamed `paymentLimit` to `amountForReceivedAmount` but the logic should be the same as it was before. I think the `paymentLimit` was misleading - it was only being used to get the asset code/scale (per your request to get it from grant limits). I also removed the typeguards (had bigger plans for them originally, but doing nothing compared to `paymentLimit.debitAmount` etc. checks now).

Reply:
[ccc2dc2](https://github.com/interledger/rafiki/pull/2753/commits/ccc2dc22ce300ec311465306324684b40a509659)

Reply:
Actually, looking at it, we can actually do something easier: I think we can use `payment.receiveAmount` -> that will have the asset code & scale of the receiver (since it uses the quote.receiveAmount) under the hood for it